packages/web/src/components/splitView.tsx

```diff
@@ -38,8 +38,14 @@ export const SplitView: React.FC<React.PropsWithChildren<SplitViewProps>> = ({
   settingName,
   children
 }) => {
-  const [hSize, setHSize] = useSetting<number>(settingName ? settingName + '.' + orientation + ':size' : undefined, Math.max(minSidebarSize, sidebarSize) * window.devicePixelRatio);
-  const [vSize, setVSize] = useSetting<number>(settingName ? settingName + '.' + orientation + ':size' : undefined, Math.max(minSidebarSize, sidebarSize) * window.devicePixelRatio);
+  const defaultSize = Math.max(minSidebarSize, sidebarSize) * window.devicePixelRatio;
+  const hSetting = useSetting<number>((settingName ?? 'unused') + '.' + orientation + ':size', defaultSize);
+  const vSetting = useSetting<number>((settingName ?? 'unused') + '.' + orientation + ':size', defaultSize);
+  const hState = React.useState(defaultSize);
+  const vState = React.useState(defaultSize);
+  const [hSize, setHSize] = settingName ? hSetting : hState;
```

Comment:
This works for me, but we could as well move this choice back into `useSetting()` helper if we ever need the conditional again. WDYT?

Reply:
Yes we could. We could even move it into a separate Hook that uses `useSetting`, so that the `useSetting` implementation isn't cluttered.